### ISSUE TITLE
InterruptStrategy not working with PHP7

### ISSUE THREAD

**Issue by basz** [juriansluiman/SlmQueue#173]
On PHP7.0.0-beta5 the InterruptStrategy is not working. So after rereading all pcntl documentation I think I have found the problem (but then again it could be just a bug in PHP). 

Apparently the declare(ticks=1) statement should go around the code you need to be interruptable. So when I modify SlmQueue/Worker/AbstractWorker in the following manner the interrupt is working again. Also I commented the declare statement in the Interrupt strategy.

```
declare( ticks=1 ) {
    while (!$workerEvent->shouldExitWorkerLoop()) {
        $eventManager->trigger(WorkerEvent::EVENT_PROCESS_QUEUE, $workerEvent);
    }
}
```

This comment helped me get this working : https://stackoverflow.com/questions/12722791/php-pcntl-signal-callbacks-are-not-called/23536312#23536312

Since this might be a beta related issue, i won't PR just yet.

**Comment by basz** [comment]
closing as PHP7.0.0-RC8 is confirmed to have fixed this problem... workers can now correctly be halted

**Comment by basz** [comment]
I'm reopening as `supervisorctl stop my-worker` is not halted my worker as it should. Running PHP7.0.0 a.t.m.

**Comment by basz** [comment]
```
php -i | grep disable_functions
disable_functions => no value => no value
```

**Comment by basz** [comment]
@juriansluiman @bakura10 have you run into this too on php 7.0.0 (fresh Ubuntu 14.04 box, ondrej ppa)? I can confirm that cntr-c won't halt my worker unless I surround to worker loop like so, but I can't help to think I'm overlooking something.

Would it hurt to PR this? Kind of annoying as my deploy scripts now wait until supervisor waits for it's maximum and kills the worker anyway.

```
declare( ticks=1 ) {
    while (!$workerEvent->shouldExitWorkerLoop()) {
        $eventManager->trigger(WorkerEvent::EVENT_PROCESS_QUEUE, $workerEvent);
    }
}
```

**Comment by bakura10** [comment]
Hi @basz ,

I have not used SlmQueue with PHP 7 yet. @juriansluiman any idea?

**Comment by basz** [comment]
@bakura are you able to manually test this on a pre PHP7. I now have now no PHP5 installs to quickly test.

**Comment by basz** [comment]
I just confirmed that the proposed PR still works on a PHP 5.5.9 installation. Can we tag this as hotfix/patch release?

**Comment by basz** [comment]
closed by #175
